spec.bs

```diff
@@ -1297,17 +1301,25 @@ by a randomized amount of time to make it difficult to determine whether a
 report was sent or not from any particular event. In the case that a
 [=pre-specified report parameters/context ID=] is supplied, a non-default
 [=pre-specified report parameters/filtering ID max bytes=] is specified, or a
-[=pre-specified report parameters/max contributions=] is specified, the API
-makes the number of reports sent deterministic (sending 'null reports' if
-necessary — each containing only a contribution with a value of 0 in the
+non-default [=pre-specified report parameters/max contributions=] is specified,
+the API makes the number of reports sent deterministic (sending 'null reports'
+if necessary — each containing only a contribution with a value of 0 in the
 payload). Additional mitigations may also be possible in the future, e.g. adding
 noise to the report count.
 
 ### Protecting against leaks via payload size ### {#protecting-against-leaks-via-payload-size}
 
-The length of the payload could additionally expose some cross-site information,
-namely how many contributions are included. To protect against this, the payload
-is padded to a fixed number of contributions.
+The length of the encrypted payload could additionally expose some cross-site
+information, namely the number of contributions present in the plaintext
+payload. To eliminate this side channel, Private Aggregation ensures that
+payloads contain a predetermined number of contributions prior to encryption,
+potentially truncating or padding with null contributions to match the target.
+
+The target number of contributions is based on [=pre-specified report
+parameters/max contributions=] when it is non-null — it is the caller's
```

Comment:
I'm not sure I understand the "it is the caller's ... cross-site data" clause. Maybe can just drop it?

Reply:
Ah, I was trying to call out that Private Aggregation assumes that `pre-specified report parameters/max contributions` is not a function of cross-site data. Maybe that's obvious? It certainly interrupts the paragraph's flow, so I'll delete it.

Reply:
Did a little more rewriting here, PTAL.

Reply:
Ah ack -- yeah we could consider expressly saying somewhere that "pre-specified report parameters" generally must be set from a first-party context. But I think it's ok as is too.